lib/puppet/defaults.rb

```diff
@@ -330,6 +336,27 @@ def self.default_diffargs
         improvement for features that are checked frequently.
       EOT
     },
+    :always_cache_misses => {
```

Comment:
I suggest reversing the behavior of the setting, and naming it to focus on the repeated checks, rather than the cache. As programmers, it's natural for you to see the cache as the most important part, but for a user, it's going to be more useful to think about it in terms of repeated work. 

Best name I can think of on the fly is `:always_retry_plugins`. (I think "plugins" is the best umbrella name for types and provider features... anyone got a better one?)